llvm/utils/TableGen/CompressInstEmitter.cpp

```diff
@@ -162,11 +163,12 @@ class CompressInstEmitter {
 } // End anonymous namespace.
 
 bool CompressInstEmitter::validateRegister(const Record *Reg,
-                                           const Record *RegClass) {
+                                           const Record *RegClass,
+                                           ArrayRef<SMLoc> Loc) {
   assert(Reg->isSubClassOf("Register") && "Reg record should be a Register");
-  assert(RegClass->isSubClassOf("RegisterClass") &&
-         "RegClass record should be a RegisterClass");
-  const CodeGenRegisterClass &RC = Target.getRegisterClass(RegClass);
+  assert(RegClass->isSubClassOf("RegisterClassLike") &&
```

Comment:
Is this change related?

Reply:
Without this change we get an assertion failure instead of an error in the second test that I added